iceoryx2-ffi/cxx/include/iox2/unique_port_id.hpp

```diff
@@ -48,6 +56,8 @@ class UniqueSubscriberId {
     auto operator=(UniqueSubscriberId&& rhs) noexcept -> UniqueSubscriberId&;
     ~UniqueSubscriberId();
 
+    auto bytes() -> iox::optional<RawIdType>&;
```

Comment:
```suggestion
    auto bytes() -> const iox::optional<RawIdType>&;
```

Reply:
Shouldn't this be then

```cxx
auto bytes() const -> const iox::optional<RawIdType>&;
```

Reply:
@elfenpiff The bytes are being copied from the Rust layer on the first call (mentioned here: https://github.com/eclipse-iceoryx/iceoryx2/pull/501#discussion_r1826542418).

I opted for the copy because I did not see how to resolve the ownership across the layers. Open for suggestions if there is a better way.

Reply:
One option I thought of was `iox::optional<const iox::Slice<u8>>` but then it felt somewhat weird and the number of copied bytes are small. Adding an indirection will most probably offset the cost of copying.

Reply:
The `UniquePortId` is an `u128` on the Rust side so a copy of bytes is completely fine and shouldn't impose any more overhead then returning a reference - it may be even faster.

But when this is a copy, either the bytes are cached internally and then returning a reference makes sense or this API leads to a dangling reference.